operations/helm/charts/grafana-agent/templates/controllers/_pod.yaml

```diff
@@ -10,7 +10,7 @@ metadata:
     {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
-  serviceAccount: {{ include "grafana-agent.serviceAccountName" . }}
+  serviceAccountName: {{ include "grafana-agent.serviceAccountName" . }}
```

Comment:
Should we check for Kubernetes API version here to choose between serviceAccount/serviceAccountName? 

The Loki Helm chart uses version checks in a few places: https://github.com/grafana/loki/blob/2c57d2e2b91a5fe8efe852350b9ca5971b73ebf3/production/helm/loki/templates/read/statefulset-read.yaml#L20

Reply:
(It depends on when serviceAccountName was first introduced. If it was introduced years ago, we probably don't need to bother to support the old field name)

Reply:
I see a reference to the deprecation back in v1.20, which was released on Dec 8, 2020 so I'd say we go ahead?

https://devdocs.io/kubernetes~1.20/workload-resources/pod-v1/index#PodSpec

Reply:
yeah, that's old enough for me, I think it's fine to just use the new field